lib/tailreadstream/tailreadstream.js

```diff
@@ -64,9 +64,10 @@ class TailReadStream extends Readable {
   get timeout() { return this._timeout }
   set timeout(timeout) { this._timeout = timeout }
 
-  destroy() {
+  _destroy() {
```

Comment:
you renamed it but where does it get called? have you changed it there?

Reply:
This class extends `Readable`, so it's a stream, and this is a mis-implementation of node streams.  The nodejs docs state that `destroy` should NOT be overridden, but it will call `_destroy` automatically if you have custom operations to be done there.  

https://nodejs.org/dist/latest-v10.x/docs/api/stream.html#stream_readable_destroy_error

```
Implementors should not override this method, but instead implement readable._destroy().
```

Reply:
The integration test also shows that it works :]. Otherwise, the process would not shut down cleanly.

Reply:
@darinspivey, is this what can potentially cause the memory leak?

Reply:
not sure, but it aint good :)

Reply:
I suppose it could because it's bypassing what the node internals do for `destroy`.